devito/propagator.py

```diff
@@ -213,6 +223,8 @@ def convert_equality_to_cgen(self, equality):
         else:
             s_lhs = ccode(self.time_substitutions(equality.lhs).xreplace(self._var_map))
             s_rhs = ccode(self.time_substitutions(equality.rhs).xreplace(self._var_map))
+            s_rhs = str(s_rhs).replace("pow", "powf")
+            s_rhs = str(s_rhs).replace("fabs", "fabsf")
```

Comment:
Yes it does, if not, there is type conversions in the code a pow return a double even if everything is a float.  There is however a dtype missing as we would need to keep pow and fabs for double precison